pkg/apis/tensorflow/v1alpha1/types.go

```diff
@@ -0,0 +1,195 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/api/core/v1"
+)
+
+const (
+	CRDKind       = "TfJob"
+	CRDKindPlural = "tfjobs"
+	CRDGroup      = "tensorflow.org"
+	CRDVersion    = "v1alpha1"
+	// Value of the APP label that gets applied to a lot of entities.
+	AppLabel = "tensorflow-job"
+	// Defaults for the Spec
+	TfPort   = 2222
+	Replicas = 1
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=tfjob
+
+// TFJob describes tfjob info
+type TfJob struct {
+	metav1.TypeMeta    `json:",inline"`
+	metav1.ObjectMeta  `json:"metadata,omitempty"`
+	Spec   TfJobSpec   `json:"spec"`
+	Status TfJobStatus `json:"status"`
+}
+
+type TfJobSpec struct {
+	// TODO(jlewi): Can we we get rid of this and use some value from Kubernetes or a random ide.
+	RuntimeId string
```

Comment:
Can we reuse metadata.UID?

Reply:
Can you open up an issue for this?

Reply:
ok

Reply:
FYI https://github.com/tensorflow/k8s/issues/249